app/javascript/role_manager.js

```diff
@@ -63,12 +69,23 @@ App.StimulusApp.register('role-manager', class extends Stimulus.Controller {
         this.showSearchPermissions(search_string, false)
       });
     } else {
+      // remove this role from the visible columns
+      this.enabledColumns = this.enabledColumns.filter(element => element !== target_role)
       this.roleColumnTargets.forEach((column) => {
         if (target_role == $(column).data('roleManagerRoleValue')) {
           $(column).addClass('hide')
         }
       });
     }
+    this.storeColumnState()
+  }
+
+  storeColumnState() {
+    window.localStorage.setItem(this.columnStateKey, JSON.stringify(this.enabledColumns));
+  }
+
+  fetchColumnState() {
+    return JSON.parse(window.localStorage.getItem(this.columnStateKey))
```

Comment:
suggest catching errors
```suggestion
try {
  return JSON.parse(window.localStorage.getItem(this.columnStateKey)) || [];
} catch (error) {
  console.error('Error parsing localStorage item:', error);
  return [];
}
```